File: suites/02_numerical_computation/comparisons/polynomial_eval.py
def polynomial_cost(points: int, degree: int) -> int:
    checksum = 0
    for p in range(points):
        x = (p % 97) + 3
        value = 1
        for c in range(degree, 0, -1):
            value = ((value * x) + ((c * 11) + (p % 29))) % 1000003
        checksum += value
    return checksum

if __name__ == "__main__":
    print(polynomial_cost(800000, 16))

File: suites/02_numerical_computation/comparisons/polynomial_eval.go
package main
import "fmt"
func polynomialCost(points, degree int64) int64 {
	var checksum int64
	for p := int64(0); p < points; p++ {
		x := (p % 97) + 3
		value := int64(1)
		for c := degree; c > 0; c-- {
			value = ((value * x) + ((c * 11) + (p % 29))) % 1000003
		}
		checksum += value
	}
	return checksum
}
func main() { fmt.Println(polynomialCost(800000, 16)) }
